src/variational/VariationalInference.jl

```diff
@@ -23,6 +23,25 @@ function __init__()
         Flux.Optimise.apply!(o::TruncatedADAGrad, x, Δ) = apply!(o, x, Δ)
         Flux.Optimise.apply!(o::DecayedADAGrad, x, Δ) = apply!(o, x, Δ)
     end
+    @require Zygote = "e88e6eb3-aa80-5325-afca-941959d7151f" begin
+        function Variational.grad!(
+            vo,
+            alg::VariationalInference{<:Turing.ZygoteAD},
+            q,
+            model,
+            θ::AbstractVector{<:Real},
+            out::DiffResults.MutableDiffResult,
+            args...
+        )
+            f(θ) = if (q isa VariationalPosterior)
+                - vo(alg, update(q, θ), model, args...)
+            else
+                - vo(alg, q(θ), model, args...)
+            end
```

Comment:
What's the reason for not defining
```suggestion
            if q isa VariationalPosterior
                f = let alg=alg, q=q, model=model, args=args
                    θ -> - vo(alg, update(q, θ), model, args...)
                end
            else
                f = let alg=alg, q=q, model=model, args=args
                    θ -> - vo(alg, q(θ), model, args...)
                end
            end
```
or at least moving the `if` statement out of the function definition? Does it compile to the same thing?

Reply:
The if statement should compile away yes. I just copied the functions from other backends' methods.

Reply:
> The if statement should compile away yes. I just copied the functions from other backends' methods.

I was the one who did this, and I indeed did it because the `if` should compile away:)